packages/common/base/src/utils/apiKey/validateAPIKeyPrefix.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, test } from "vitest";
+
+import { validateAPIKeyPrefix } from "./validateAPIKeyPrefix";
+
+describe("validateAPIKeyPrefix", () => {
+    test("Should disallow API keys with old format", () => {
+        const prefixes = ["sk_live", "sk_test"];
+
+        for (const prefix of prefixes) {
+            const result = validateAPIKeyPrefix(prefix);
+            if (result.isValid) {
+                throw new Error("Expected API key to be invalid");
+            }
+
+            expect(result.isValid).toBe(false);
+            expect(result.message).toBe(
+                "Old API key format detected. Please create a new API key from the Crossmint console."
+            );
+        }
+    });
+
+    test("Should disallow when usage prefix is invalid", () => {
+        const result = validateAPIKeyPrefix("bk_");
+        if (result.isValid) {
+            throw new Error("Expected API key to be invalid");
+        }
+
+        expect(result.isValid).toBe(false);
+        expect(result.message).toBe("Malformed API key. Must start with 'ck' or 'sk'.");
+    });
+
+    test("Should disallow when API key does not have expected usage origin", () => {
+        const keys = ["ck_development", "sk_development"];
+
+        for (const key of keys) {
+            const usageOrigin = key.startsWith("ck") ? "client" : "server";
+            const expectedUsageOrigin = usageOrigin === "client" ? "server" : "client";
+
+            const result = validateAPIKeyPrefix(key, { usageOrigin: expectedUsageOrigin });
+            if (result.isValid) {
+                throw new Error("Expected API key to be invalid");
+            }
+
+            expect(result.isValid).toBe(false);
+            expect(result.message).toBe(
+                `Disallowed API key. You passed a ${usageOrigin} API key, but a ${expectedUsageOrigin} API key is required.`
+            );
+        }
```

Comment:
it feels like we've made the code more complicated so that we can use a for loop on an array of 2 variables
```suggestion
        const clientDevKey = "ck_development"
        const result = validateAPIKeyPrefix(key, { usageOrigin: "server" });
        expect(result.isValid).toBe(false);
        expect(result.message).toBe(
            `Disallowed API key. You passed a ${"client"} API key, but a ${"server"} API key is required.`
        );
    
        const serverDevKey = "sk_development"
        const result = validateAPIKeyPrefix(key, { usageOrigin: "client" });
        expect(result.isValid).toBe(false);
        expect(result.message).toBe(
            `Disallowed API key. You passed a ${"server"} API key, but a ${"client"} API key is required.`
        );
```
even fewer lines and way more readable